packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -334,6 +335,38 @@ const ProductDetail = () => {
         }))
     }
 
+    // Transfer pickup state when variant changes if inventory available, otherwise clear
```

Comment:
Thinking a bit about this solution, from what I understand is that from each variant, shopper can choose to do pick up at store or delivery to address. the pickupInStoreMap is to save these state for each variant e.g 
```
psedo-data
{variant-1: {method: 'pickup'}, variant-2: {'method' 'delivery'}}. 
```
I fine it is not a good reason to keep transfering state from one variant to another since we already give the shopper freedom to choose on each variant. Let the delivery options state reset on each variant if it is not selected (NOT existing in pickupInStoreMap), if it is selected, then we show the selected option
If we want to do the transfer, I think we should always keep one delivery option state per PDP for all variants (which I think less flexible than current implementation)